README.md

```diff
@@ -1,368 +1,257 @@
 
-### Helper Method
-The main API is a helper:
+### Globally Exposing Your React Components
+Place your JavaScript code inside of the provided `client/app` folder. Use modules just as you would when using webpack alone. The difference here is that instead of mounting React components directly to an element using `React.render`, you **expose your components globally and then mount them with helpers inside of your Rails views**.
 
-```ruby
-  <%= react_component(component_name, props = {}, options = {}) %>
-```
-  
-Params are:
++ *Normal Mode (JavaScript is Rendered on client):*
 
-* **react_component_name**: [string] can be a React component, created using a ES6 class, or React.createClass,
-  or a `generator function` that returns a React component
-   
-  using ES6
   ```javascript
-  let MyReactComponentApp = (props) => <MyReactComponent {...props}/>;
-  ```       
-     
-  or using ES5
+  window.HelloWorld = HelloWorldAppClient;
+  ```
++ *Server-Side Rendering:*
+
   ```javascript
-  var MyReactComponentApp = function(props) { return <YourReactComponent {...props}/>; }
+  global.HelloWorld = HelloWorldAppServer;
   ```
-  Exposing the react_component_name is necessary to both a plain ReactComponent as well as
-     a generator:
-   For client rendering, expose the react_component_name on window:
    
-  ```javascript
-  window.MyReactComponentApp = MyReactComponentApp;
-  ```       
-  For server rendering, export the react_component_name on global:
-  ```javascript
-  global.MyReactComponentApp = MyReactComponentApp;
-  ```       
-  If you're curious as to what the gem generates for the server and client rendering, see [`spec/dummy/client/app/startup/serverGlobals.jsx`](https://github.com/shakacode/react_on_rails/blob/master/spec/dummy/spec/sample_generated_js/server-generated.js)
-  and [`spec/dummy/client/app/startup/ClientReduxApp.jsx`](https://github.com/shakacode/react_on_rails/blob/master/spec/dummy/spec/sample_generated_js/client-generated.js) for examples of this. Note, this is not the code that you are providing. You can see the client code by viewing the page source.
-  
-* **props**: [hash | string of json] Properties to pass to the react object. See this example if you're using Jbuilder: [react-webpack-rails-tutorial view rendering props using jBuilder](https://github.com/shakacode/react-webpack-rails-tutorial/blob/master/app/views/pages/index.html.erb#L20)
+### Including Your JavaScript in Rails Views
+Once the bundled files have been generated in your `app/assets/javascripts/generated` folder and you have exposed your components globally, you will want to run your code in your Rails views using the included helper method (for more on the helper methods, [see below](#helpers)).
+
++ *Normal Mode (JavaScript is rendered on client):*
+
+  ```erb
+  <%= react_component("HelloWorldApp") %>
```

Comment:
show both with same options and use @some_props as the variable name with a comment that 

@props can be either a hash or JSON string

the {} is too vague